docs/architecture.md

```diff
@@ -0,0 +1,185 @@
+Starting with Pydantic V2, part of the codebase is written in Rust in a separate package called `pydantic-core`.
+This architecture documentation will partly focus on how the two `pydantic` and `pydantic-core` packages interacts together.
```

Comment:
Maybe something that touches on a few of the other things covered, like:

This architecture documentation will first cover how `pydantic` and `pydantic-core` interact, then will delve into architecture specifics for various patterns, such as model definition, validation, and serialization.